spec/requests/OrganisationsController/organisations_controller_rent_periods_spec.rb

```diff
@@ -0,0 +1,114 @@
+require "rails_helper"
+
+RSpec.describe OrganisationsController, type: :request do
+  let(:user) { create(:user, :support) }
+  let(:headers) { { "Accept" => "text/html" } }
+  let(:page) { Capybara::Node::Simple.new(response.body) }
+
+  before do
+    allow(user).to receive(:need_two_factor_authentication?).and_return(false)
+    sign_in user
+  end
+
+  describe "#new" do
+    before do
+      get new_organisation_path
+    end
+
+    it "displays the rent periods question" do
+      expect(page).to have_content "What are the rent periods for the organisation?"
+    end
+
+    it "the checkboxes for each rent period are checked by default" do
+      checkboxes = page.all "input[type='checkbox'][name='organisation[rent_periods][]']"
+      expect(checkboxes.count).to be > 5
```

Comment:
interested in thoughts on this. It kind of looks random to be greater than 5 (which is an arbitrary number) but it's annoying when this type of test fails because the number of options changes by 1 due to something unrelated.

I think it's important to have something, otherwise the next line can pass when there are no checkboxes on the page.

maybe I should just change it to > 0 ? Or maybe it's good as is.

Reply:
How likely would we add new rent periods?

Reply:
in what time frame?
low but not zero.
is this test currently doing its job and will it fail in future for good reason or no reason?

part of the reason why I've done this is the real expectation is the next line on 25, but any time you check all of something meet a condition the test can fail to be testing what it's meant to if there are none of that thing on the page because something earlier in the test setup gets broken at some point in the future so the test silently passes despite not doing what it's meant to